tools/x-docs-2/src/components/sidebar/module-list.jsx

```diff
@@ -1,26 +1,28 @@
 import React from 'react';
 import { Link } from 'gatsby';
 
+import GithubSlugger from 'github-slugger';
+
 const createUrlFromName = (name) => {
-	return name.toLowerCase().replace(/ /g, '-').replace(/[?!:%$'.,"…:]/g, '');
+	const slugger = new GithubSlugger();
```

Comment:
if you hoist this into module scope you'll only need to contruct a single instance =]

Reply:
The problem is that when it constructs an instance and then applies it to same string it checks it for uniqueness, like this:
```
const slugger = new GithubSlugger();
slugger.slug('url one') // url-one
slugger.slug('url one') // url-one-2
```
So we need to create a new instance every time =(

Reply:
Ah OK, definitely worth adding a comment to explain this!